### PR TITLE
feat(agent): get / store enabled tasks from agent

### PR DESCRIPTION
**Pull request by stonebuzz**
Handle enabled modules from agent.

image

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: install/migrations/update_10.0.3_to_10.0.4/inventory.php [hunk 1]
```diff
@@ -62,3 +62,13 @@
...
+
+//add new fields for Agent
+$migration->addField("glpi_agents", 'use_module_wake_on_lan', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_computer_inventory', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_esx_remote_inventory', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_remote_inventory', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_network_inventory', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_network_discovery', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_package_deployment', 'bool', ['value' => 0]);
+$migration->addField("glpi_agents", 'use_module_collect_data', 'bool', ['value' => 0]);
```
Comment (cedric-anne) [inline install/migrations/update_10.0.3_to_10.0.4/inventory.php:74]: `0` is already the default value.
```suggestion
$migration->addField("glpi_agents", 'use_module_wake_on_lan', 'bool');
$migration->addField("glpi_agents", 'use_module_computer_inventory', 'bool');
$migration->addField("glpi_agents", 'use_module_esx_remote_inventory', 'bool');
$migration->addField("glpi_agents", 'use_module_remote_inventory', 'bool');
$migration->addField("glpi_agents", 'use_module_network_inventory', 'bool');
$migration->addField("glpi_agents", 'use_module_network_discovery', 'bool');
$migration->addField("glpi_agents", 'use_module_package_deployment', 'bool');
$migration->addField("glpi_agents", 'use_module_collect_data', 'bool');
```

**Review by cedric-anne — APPROVED** [review]

File: templates/pages/admin/inventory/agent.html.twig [hunk 2]
```diff
@@ -146,5 +148,25 @@
...
+            'use_module_wake_on_lan': __('Wake on LAN'),
+            'use_module_computer_inventory': __('Computer inventory'),
+            'use_module_esx_remote_inventory': __('ESX remote inventory'),
+            'use_module_network_inventory': __('Network inventory (SNMP)'),
+            'use_module_network_discovery': __('Network discovery (SNMP)'),
+            'use_module_package_deployment': __('Package Deployment'),
+            'use_module_collect_data': __('Collect data'),
+            'use_module_remote_inventory': __('Remote inventory'),
+        } %}
+            {{ fields.checkboxField(
```
Comment (orthagh) [inline templates/pages/admin/inventory/agent.html.twig:164]: rather a checkbox field which can mislead user (there is maybe a way to enable/disable the module), a `<i class="ti-ti-check">`

Comment (stonebuzz) [reply templates/pages/admin/inventory/agent.html.twig:164]: image

**Review by trasher — APPROVED** [review]

File: templates/pages/admin/inventory/agent.html.twig [hunk 3]
```diff
@@ -146,5 +148,24 @@
...
+            'use_module_package_deployment': __('Package Deployment'),
+            'use_module_collect_data': __('Collect data'),
+            'use_module_remote_inventory': __('Remote inventory'),
+        } %}
+
+        {% if not item.fields[modulefield] %}
+            {% set html = '<i class="ti ti-check"></i>' %}
+        {% endif %}
+        {{ fields.htmlField('', html, modulelabel) }}
+        {% endfor %}
```
Comment (cedric-anne) [inline templates/pages/admin/inventory/agent.html.twig:169]: 1. `html` value will inherit from previous item in loop if condition does not match.
2. Are you sure that you want to display a check on `NOT` condition ?

**Pull request merged**